perfkitbenchmarker/sample.py

```diff
@@ -57,12 +58,57 @@ class Sample(collections.namedtuple('Sample', _SAMPLE_FIELDS)):
     value: float. Result for 'metric'.
     unit: string. Units for 'value'.
     metadata: dict. Additional metadata to include with the sample.
+    timestamp: float. Unix timestamp.
   """
 
-  def __new__(cls, metric, value, unit, metadata=None, **kwargs):
+  def __new__(cls, metric, value, unit, metadata=None, timestamp=None,
+              **kwargs):
+    if timestamp is None:
+      timestamp = time.time()
+
     return super(Sample, cls).__new__(cls, metric, value, unit,
-                                      metadata=metadata or {}, **kwargs)
+                                      metadata=metadata or {},
+                                      timestamp=timestamp,
+                                      **kwargs)
 
   def asdict(self):
     """Converts the Sample to a dictionary."""
     return self._asdict()
+
+
+class SamplesTestMixin(object):
+  """A mixin for unittest.TestCase that adds a type-specific equality
+  predicate for samples.
+  """
+
+  def __init__(self, *args, **kwargs):
+    super(SamplesTestMixin, self).__init__(self, *args, **kwargs)
+
+    self.addTypeEqualityFunc(Sample, self.assertSamplesEqual)
+
+  def assertSamplesEqual(self, a, b, msg=None):
```

Comment:
Could you add a docstring? At minimum this method should state that timestamp is ignored. Consider also renaming it to make it clear that the timestamp is being ignored.